FluentAssertions.Core/Types/TypeAssertions.cs

```diff
@@ -198,6 +200,366 @@ public AndConstraint<TypeAssertions> BeDecoratedWith<TAttribute>(string because
+                .BecauseOf(because, reasonArgs)
+                .FailWith(String.Format("Expected {0} {1}.{2} to exist{{reason}}, but it does not.",
+                    propertyType.Name, Subject.FullName, name));
+
+            Execute.Assertion.ForCondition(propertyInfo.PropertyType == propertyType)
+                .BecauseOf(because, reasonArgs)
+                .FailWith(String.Format("Expected {0} to be of type {1}{{reason}}, but it is not.",
+                    propertyInfoDescription, propertyType));
+
+            return new AndWhichConstraint<TypeAssertions, PropertyInfo>(this, propertyInfo);
+        }
+
+        /// <summary>
+        /// Asserts that the current type does not have a property named <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The name of the property.</param>
+        /// <param name="because">A formatted phrase as is supported by <see cref="M:System.String.Format(System.String,System.Object[])"/> explaining why the assertion
+        ///             is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.</param>
+        /// <param name="reasonArgs">Zero or more objects to format using the placeholders in <see cref="!:because"/>.</param>
+        public AndConstraint<TypeAssertions> NotHaveProperty(string name, string because = "", params object[] reasonArgs)
+        {
+            PropertyInfo propertyInfo = Subject.GetPropertyByName(name);
+
+            string propertyInfoDescription = "";
+
+            if (propertyInfo != null)
+            {
+                propertyInfoDescription = PropertyInfoAssertions.GetDescriptionFor(propertyInfo);
+            }
+
+            Execute.Assertion.ForCondition(propertyInfo == null)
+                .BecauseOf(because, reasonArgs)
+                .FailWith(string.Format("Expected {0} to not exist{{reason}}, but it does.", propertyInfoDescription));
+
+            return new AndConstraint<TypeAssertions>(this);
+        }
+
+        /// <summary>
+        /// Asserts that the current type explicitly implements a property named 
+        /// <paramref name="name"/> from interface <paramref name="interfaceType" />.
+        /// </summary>
+        /// <param name="interfaceType">The type of the interface.</param>
+        /// <param name="name">The name of the property.</param>
+        /// <param name="because">A formatted phrase as is supported by <see cref="M:System.String.Format(System.String,System.Object[])"/> explaining why the assertion
+        ///     is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.</param>
+        /// <param name="reasonArgs">Zero or more objects to format using the placeholders in <see cref="!:because"/>.</param>
+        public AndConstraint<TypeAssertions> HaveExplicitProperty(Type interfaceType, string name, string because = "", params object[] reasonArgs)
```

Comment:
is interfaceType required? It feels like you can do that without interface type. I'd rename current method to ExplicitlyImplementProperty